tests/libvmifact/factory.go

```diff
@@ -88,16 +88,22 @@ func NewAlpineWithTestTooling(opts ...libvmi.Option) *kvirtv1.VirtualMachineInst
 	vmi := libvmi.New(alpineOpts...)
 
 	// Supplied with no user data, AlpimeWithTestTooling image takes more than 200s to allow login
-	if !cloudInitNoCloudExists(vmi.Spec.Volumes) {
+	if !cloudInitExists(vmi.Spec.Volumes) {
 		withNonEmptyUserData := libvmi.WithCloudInitNoCloud(libvmici.WithNoCloudEncodedUserData("#!/bin/bash\necho hello\n"))
 		withNonEmptyUserData(vmi)
 	}
 	return vmi
 }
 
-func cloudInitNoCloudExists(volumes []kvirtv1.Volume) bool {
+func cloudInitExists(volumes []kvirtv1.Volume) bool {
```

Comment:
What about if someone calls new cirros with cloud init configdrive with network data, it will pass this check and then will run over the volume with new volumesource nocloud with the user data.
And also previously some functions calls the functions we changed to the options that put network data and userdata="". now if you call newcirros with network data nocloud only, you will still add the userdata which I dont know if thats the desired behavior..

Reply:
If I read the code correctly, it won't override the volume but will add fields to it

Reply:
you cant have different volumesources for the same volume..
and what about the second point of do we want to add userdata in case of requested network data?

Reply:
@ShellyKa13 

What if we'll do:
```go
func cloudInitExists(volumes []kvirtv1.Volume) bool {
	for _, vol := range volumes {
		if noCloudSrc := vol.CloudInitNoCloud; noCloudSrc != nil &&
			(noCloudSrc.UserData != "" || noCloudSrc.UserDataBase64 != "" || noCloudSrc.UserDataSecretRef != nil) {
			return true
		}

		if vol.CloudInitConfigDrive != nil {
			return true
		}
	}
	return false
}

```

Reply:
We can maybe even do:

```go
func cloudInitExists(volumes []kvirtv1.Volume) bool {
	for _, vol := range volumes {
		if vol.CloudInitNoCloud != nil {
			return true
		}

		if vol.CloudInitConfigDrive != nil {
			return true
		}
	}
	return false
}
```

Reply:
@EdDev - WDYT?

Reply:
sounds like its enough to check  if volumeExists(vmi, cloudInitDiskName)
(need to make cloudInitDiskName public)